### PR TITLE
Adding labels to test-operator resources

### PR DESCRIPTION
**Pull request by kstrenkova**

### CODE REVIEW

File: pkg/tempest/job.go [hunk 1]
```diff
@@ -31,6 +31,9 @@ func Job(
 		Spec: batchv1.JobSpec{
 			BackoffLimit: instance.Spec.BackoffLimit,
 			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
```
Comment (lpiwowar) [inline pkg/tempest/job.go:34]: +1

File: pkg/tobiko/job.go [hunk 2]
```diff
@@ -36,6 +36,9 @@ func Job(
 			Completions:  &completions,
 			BackoffLimit: instance.Spec.BackoffLimit,
 			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
```
Comment (lpiwowar) [inline pkg/tobiko/job.go:39]: +1

File: controllers/tobiko_controller.go [hunk 3]
```diff
@@ -132,13 +153,72 @@ func (r *TobikoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 
-	result, err := r.EnsureTobikoCloudsYAML(ctx, instance, helper)
+	// Prepare env vars
```
Comment (lpiwowar) [inline controllers/tobiko_controller.go:156]: Why was this section added here?

File: controllers/tobiko_controller.go [hunk 4]
```diff
@@ -71,12 +71,33 @@ func (r *TobikoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
+	logging := log.FromContext(ctx)
+	if r.CompletedJobExists(ctx, instance) {
+		// The job created by the instance was completed. Release the lock
+		// so that other instances can spawn a job.
+		logging.Info("Job completed")
+		r.ReleaseLock(ctx, instance)
+	}
```
Comment (lpiwowar) [inline controllers/tobiko_controller.go:80]: Why do we need to add this section of the code?

File: controllers/tobiko_controller.go [hunk 5]
```diff
@@ -71,12 +71,33 @@ func (r *TobikoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
...
 	workflowActive := false
 	if len(instance.Spec.Workflow) > 0 {
 		workflowActive = true
 	}
 
+	// Combine labels
+	for key, value := range operatorLabels {
+		serviceLabels[key] = value
+	}
+
```
Comment (lpiwowar) [inline controllers/tobiko_controller.go:100]: Do we need to combine the `operatorLabels` and `serviceLabels` like this? Can't we do something like:
```
serviceLabels := map[string]string {
  "operator": "test-operator",
  "test-name": instance.Name,
  common.AppSelector: tobiko.ServiceName
}
```

File: controllers/tempest_controller.go [hunk 6]
```diff
@@ -188,10 +188,20 @@ func (r *TempestReconciler) reconcileNormal(ctx context.Context, instance *testv
...
+
 	serviceLabels := map[string]string{
 		common.AppSelector: tempest.ServiceName,
 	}
 
+	// Combine labels
+	for key, value := range operatorLabels {
+		serviceLabels[key] = value
+	}
+
```
Comment (lpiwowar) [inline controllers/tempest_controller.go:204]: Do we need to combine the operatorLabels and serviceLabels like this? Can't we do something like:
```
serviceLabels := map[string]string {
  "operator": "test-operator",
  "test-name": instance.Name,
  common.AppSelector: tobiko.ServiceName
}
```

File: controllers/common.go [hunk 7]
```diff
@@ -75,6 +75,7 @@ func (r *Reconciler) EnsureLogsPVCExists(
 	instance client.Object,
 	helper *helper.Helper,
 	NamePVC string,
+	labels map[string]string,
```
Comment (lpiwowar) [inline controllers/common.go:78]: +1

**Comment by lpiwowar** [comment]
So far looks good to me! Thank you @kstrenkova!

I've just added a couple of comments. We can go through them together on Wednesday. I think they can be resolved relatively quickly. Also, I think for this PR it would be great to have a single commit. You can squash the commits together using for example interactive rebase.

File: controllers/tobiko_controller.go (hunk 4 above)
Comment (kstrenkova) [reply controllers/tobiko_controller.go:80]: This section was created by merging some conflicts. I don't know if it's needed, I added it just to be sure I don't erase something that should be left here.

**Review by lpiwowar — APPROVED** [review]

**Pull request merged**